### PR TITLE
Add servers component

### PR DESCRIPTION
**Pull request by JoramD0**
**When merged this pull request will:**
- Add 2 main menu spotlight entries for the contracts and training servers (getting the password from CBA).

Note:
CBA only saves passwords entered in the server browser. That means that if someone joined through Swifty or entered the password after clicking the menu spotlight button it will not save the password for next time

### CODE REVIEW

File: addons/servers/$PBOPREFIX$ [hunk 1]
```diff
@@ -0,0 +1 @@
+x\tac\addons\servers
```
Comment (jonpas) [inline addons/servers/$PBOPREFIX$:1]:
```suggestion
x\tac\addons\servers

```

File: addons/servers/CfgMainMenuSpotlight.hpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class CfgMainMenuSpotlight
+{
```
Comment (jonpas) [inline addons/servers/CfgMainMenuSpotlight.hpp:2]: Opening bracket on same line!

File: addons/servers/CfgMainMenuSpotlight.hpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    class JoinContractServer
+    {
+        text = CSTRING(contractsText);
+        picture = QPATHTOF(ui\spotlight_theseus.paa);
+        action = "private _passwordCache = profileNamespace getVariable ['cba_ui_ServerPasswords', [[], []]];  private _index = (_passwordCache select 0) find '213.239.214.39:2302';  private _password = (_passwordCache select 1) param [_index, ''];  connectToServer ['213.239.214.39', 2302, _password];";
```
Comment (jonpas) [inline addons/servers/CfgMainMenuSpotlight.hpp:7]: Let's make this into a function. Can re-use same code for both servers then.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/servers/XEH_preInit.sqf [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+ADDON = false;
+
+PREP_RECOMPILE_START;
+#include "XEH_PREP.hpp"
+PREP_RECOMPILE_END;
+
+GVAR(vehiclesLoaded) = false;
+GVAR(isDebug) = false;
+GVAR(vehiclesList) = [];
```
Comment (jonpas) [inline addons/servers/XEH_preInit.sqf:11]: Remove those 3.

File: addons/servers/XEH_preInit.sqf [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+ADDON = true;
```
Comment (jonpas) [inline addons/servers/XEH_preInit.sqf:13]:
```suggestion
ADDON = true;

```

File: addons/servers/config.cpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        requiredAddons[] = {"tac_main"};
+        author = ECSTRING(main,Author);
+        authors[] = {"JoramD"};
+        url = ECSTRING(main,URL);
+        VERSION_CONFIG;
+    };
+};
+
+#include "CfgEventHandlers.hpp"
+
```
Comment (jonpas) [inline addons/servers/config.cpp:18]:
```suggestion
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/servers/functions/fnc_joinServer.sqf [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * Example:
+ * ["213.239.214.39", 2302] call tac_servers_fnc_joinServer
+ *
+ * Public: No
+ */
+
+params ["_ip", "_port"];
+
+private _passwordCache = profileNamespace getVariable ["cba_ui_ServerPasswords", [[], []]];
+
```
Comment (jonpas) [inline addons/servers/functions/fnc_joinServer.sqf:22]:
```suggestion
```

File: addons/servers/functions/fnc_joinServer.sqf [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+_passwordCache params ["_addresses", "_passwords"];
+
+private _index = _addresses find (format ["%1:%2", _ip, _port]);
+
```
Comment (jonpas) [inline addons/servers/functions/fnc_joinServer.sqf:26]:
```suggestion
```

File: addons/servers/CfgMainMenuSpotlight.hpp (hunk 2 above)
Comment (jonpas) [reply addons/servers/CfgMainMenuSpotlight.hpp:2]: Not done!

**Review by jonpas — CHANGES_REQUESTED** [review]

**Review by jonpas — APPROVED** [review]

File: addons/servers/CfgMainMenuSpotlight.hpp [hunk 9]
```diff
@@ -0,0 +1,16 @@
+class CfgMainMenuSpotlight {
+    class JoinContractServer {
```
Comment (jonpas) [inline addons/servers/CfgMainMenuSpotlight.hpp:2]:
```suggestion
    class GVAR(joinContracts) {
```

File: addons/servers/CfgMainMenuSpotlight.hpp [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+        text = CSTRING(contractsText);
+        picture = QPATHTOF(ui\spotlight_theseus.paa);
+        action = '["213.239.214.39", 2302] call (uiNamespace getVariable QFUNC(joinServer))';
+        actionText = CSTRING(contractsActionText);
+        condition = "true";
+    };
+    class JoinTrainingServer {
```
Comment (jonpas) [inline addons/servers/CfgMainMenuSpotlight.hpp:9]:
```suggestion
    class GVAR(joinTraining) {
```

**Review by jonpas — CHANGES_REQUESTED** [review]

**Review by jonpas — APPROVED** [review]

**Comment by jonpas** [comment]
@JoramD0 tested latest?

**Comment by JoramD0** [comment]
Not yet, on my list for today

**Comment by JoramD0** [comment]
Looks like it is all working as intended

**Review by jonpas — APPROVED** [review]

**Pull request merged**